### PR TITLE
fix(ci): update Lotus in devnets

### PR DESCRIPTION
**Pull request by LesnyRumcajs**
## Summary of changes

Changes introduced in this pull request:

- update the Lotus in devnet to address https://github.com/filecoin-project/lotus/issues/12467 that appeared in https://github.com/ChainSafe/forest/actions/runs/10940044994/job/30371863377
- address some random issues with the scripts,
- legacy drand mainnet is no longer supported in Lotus, so it doesn't make sense to support it in Forest.

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation. All new code adheres to the team's [documentation standards](https://www.notion.so/chainsafe/Documentation-practices-1a7ec5e5486b4a56b26d9df6b1677a63),
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by hanabi1224 — APPROVED** [review]

File: scripts/devnet/lotus.dockerfile [hunk 1]
```diff
@@ -19,7 +19,7 @@ RUN CGO_CFLAGS_ALLOW="-D__BLST_PORTABLE__" \
     FFI_USE_GPU="0" \
     make 2k && strip lotus*
 
-FROM ubuntu:22.04
+FROM ubuntu:24.04
```
Comment (hanabi1224) [inline scripts/devnet/lotus.dockerfile:22]: I think `ubuntu:24.04` needs to be paired with `FROM golang:1.21-bookworm AS lotus-builder`

File: scripts/devnet/setup.sh [hunk 2]
```diff
@@ -26,8 +26,8 @@ docker compose up --build --force-recreate --detach
 # RPC server is up. This checks if Forest's RPC endpoint is up.
  function call_forest_chain_head {
     curl --silent -X POST -H "Content-Type: application/json" \
-         --data '{"jsonrpc":"2.0","id":2,"method":"Filecoin.NetPeers","params":"null"}' \
-         "http://127.0.0.1:${FOREST_RPC_PORT}/rpc/v0"
+         --data '{"jsonrpc":"2.0","id":2,"method":"Filecoin.ChainHead","param":"null"}' \
```
Comment (ruseinov) [inline scripts/devnet/setup.sh:29]: Curious what happened here, just a copy-pasta mistake previously?

File: src/networks/devnet/mod.rs [hunk 3]
```diff
@@ -159,17 +155,11 @@ fn get_bundle_cid(version: &str) -> Cid {
...
-        },
-        DrandPoint {
-            height: get_upgrade_height_from_env("FOREST_DRAND_QUICKNET_HEIGHT").unwrap_or(i64::MAX),
-            config: &DRAND_QUICKNET,
-        },
-    ]
+pub(super) static DRAND_SCHEDULE: Lazy<[DrandPoint<'static>; 1]> = Lazy::new(|| {
+    [DrandPoint {
+        height: 0,
+        config: &DRAND_QUICKNET,
```
Comment (ruseinov) [inline src/networks/devnet/mod.rs:161]: Just to clarify: `DRAND_MAINNET` has been removed, because it's not needed in DEVNET setup, right?

Comment (LesnyRumcajs) [reply src/networks/devnet/mod.rs:161]: See the PR description.

File: scripts/devnet/setup.sh (hunk 2 above)
Comment (LesnyRumcajs) [reply scripts/devnet/setup.sh:29]: Likely.

File: f3-sidecar/Dockerfile [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # F3 sidecar binaries image
-FROM golang:1.22-bullseye AS builder
+FROM golang:1.22-bookworm AS builder
```
Comment (hanabi1224) [inline f3-sidecar/Dockerfile:2]: I think the `ubuntu:focal` below should be changed to `ubuntu:noble`

File: src/networks/devnet/mod.rs [hunk 5]
```diff
@@ -140,10 +139,7 @@ pub static HEIGHT_INFOS: Lazy<HashMap<Height, HeightInfo>> = Lazy::new(|| {
             get_upgrade_height_from_env("FOREST_DRAGON_HEIGHT").unwrap_or(20),
             get_bundle_cid("v13.0.0")
         ),
-        make_height!(
-            Phoenix,
-            get_upgrade_height_from_env("FOREST_DRAND_QUICKNET_HEIGHT").unwrap_or(i64::MAX)
```
Comment (hanabi1224) [inline src/networks/devnet/mod.rs:145]: `FOREST_DRAND_QUICKNET_HEIGHT` env var should be gone for good @ruseinov

**Review by ruseinov — APPROVED** [review]

**Review by hanabi1224 — APPROVED** [review]

**Pull request merged**